### PR TITLE
rancher-cifs, Microsoft Azure file storage driver

### PR DESCRIPTION
**Pull request by mfar-bcgdv**
https://github.com/rancher/storage/pull/53

### CODE REVIEW

**Review by dominicporteous — APPROVED** [review]

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/9206

File: infra-templates/cifs/0/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    pid: host
+    labels:
+      io.rancher.scheduler.global: 'true'
+      io.rancher.container.create_agent: 'true'
+      io.rancher.container.agent.role: environment
+      io.rancher.scheduler.affinity:host_label: ${HOST_LABEL}
+    environment:
+      CIFS_SERVER: '${CIFS_SERVER}'
+      MOUNT_DIR: '${MOUNT_DIR}'
+      MOUNT_OPTS: '${CIFS_VERS},user=${USER},password=${PASSWORD},${MOUNT_OPTS}'
```
Comment (LLParse) [inline infra-templates/cifs/0/docker-compose.yml:15]: We should add `RANCHER_DEBUG: '${RANCHER_DEBUG}'` so the debug flag functions (some of the other drivers are missing this flag as well...)

File: infra-templates/cifs/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+version: '2'
+services:
+  cifs-driver:
+    privileged: true
+    image: rancher/storage-cifs:v0.1
```
Comment (LLParse) [inline infra-templates/cifs/0/docker-compose.yml:5]: This will need to be modified once we release the storage plugin

**Comment by LLParse** [comment]
@deniseschannon We should consider renaming this plugin from `rancher-cifs` to `rancher-azurefs` or similar because the hard-coded configuration options are specific to Azure File Service.

File: infra-templates/cifs/0/rancher-compose.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      default: false
+      options:
+      - true
+      - false
+    - variable: HOST_LABEL
+      label: Host Label
+      type: string
+      description: Host label to schedule container
+      default: role=platform-storage
+    - variable: SCOPE
```
Comment (LLParse) [inline infra-templates/cifs/0/rancher-compose.yml:53]: Scope and Volume Access Mode really shouldn't be configurable. We should just hard-code them to environment/multiHostRW, respectively.

**Review by LLParse — CHANGES_REQUESTED** [review]
Just a few minor tweaks, but otherwise this is looking good.

**Comment by Daxten** [comment]
hey this looks quite usefull for me, is there still work required here?

**Comment by alena1108** [comment]
With the release of Rancher 2.0, development on v1.6 is only limited to critical bug fixes and security patches

**Pull request closed without merging**